### PR TITLE
Fix #1850: Add package hierarchy in testsuite.javalib.

### LINKED ISSUE #1850: Add sub-package structure in javalib testSuite.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd.

File: test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java [hunk 1]
```diff
@@ -5,12 +5,14 @@
 ** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
 **                          |/____/                                     **
 \*                                                                      */
-package org.scalajs.testsuite.javalib
+package org.scalajs.testsuite.javalib.util
 
 import java.{util => ju}
```
Comment (sjrd) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: What are those red rectangles, here?

Comment (nicolasstucki) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: I have no idea. I made sure that the character where exactly the same as the ones that were there before the commit.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/util/stream/StreamsTest.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@
 ** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
 **                          |/____/                                     **
 \*                                                                      */
-package org.scalajs.testsuite.javalib
+package org.scalajs.testsuite.javalib.util.stream
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/util/stream/StreamsTest.scala]: Hum, this should be `javalib.io`, shouldn't it?

File: test-suite/src/test/scala/org/scalajs/testsuite/typedarray/ArrayBufferInputStreamTest.scala [hunk 3]
```diff
@@ -9,7 +9,7 @@ package org.scalajs.testsuite.typedarray
 
 import scala.scalajs.js.typedarray._
 
-import org.scalajs.testsuite.javalib.CommonStreamsTests
+import org.scalajs.testsuite.javalib.util.stream.CommonStreamsTests
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/typedarray/ArrayBufferInputStreamTest.scala]: And hence, this too.

File: test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java (hunk 1 above)
Comment (sjrd) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: I suggest to use this opportunity to fix the spaces into normal spaces ;) Boy Scout rule.

**Comment by sjrd** [comment]
The commit message should say "package hierarchy", not "class hierarchy"

**Comment by sjrd** [comment]
That's all.

File: test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java (hunk 1 above)
Comment (nicolasstucki) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: The weird thing is that those are normal spaces based on `git diff`.

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/util/stream/StreamsTest.scala (hunk 2 above)
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/util/stream/StreamsTest.scala]: Yes, got confused with JDK8 streams.

**Comment by nicolasstucki** [comment]
Retest this please

**Comment by nicolasstucki** [comment]
retest this please

**Comment by nicolasstucki** [comment]
Jenkins, retest this please

File: test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java (hunk 1 above)
Comment (sjrd) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: So, did you fix those and they still appear red on GitHub for some reason?

Comment (nicolasstucki) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: I can't re wort the line but `git diff` does not see a change and as such i can't really commit.

Comment (sjrd) [inline test-suite/src/test/require-jdk7/org/scalajs/testsuite/javalib/util/CollectionsTestOnJDK7.java]: Ah ... OK never mind, then.

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
Oops, this PR contained mistaken .java files :-s I de-merged it with a force-push on master.

**Pull request merged**